File: apps/cli/src/git.zig
const std = @import("std");

pub fn getStagedChanges(allocator: std.mem.Allocator) ![]u8 {
    const result = try std.process.Child.run(.{ .allocator = allocator, .argv = &[_][]const u8{
        "git",
        "diff",
        "--cached",
    } });

    if (result.stdout.len == 0) {
        try std.io.getStdOut().writer().print("No changes to commit\n", .{});
        std.process.exit(0);
    }

    return result.stdout;
}

pub fn commit(allocator: std.mem.Allocator, message: []const u8) !void {
    _ = try std.process.Child.run(.{ .allocator = allocator, .argv = &[_][]const u8{
        "git",
        "commit",
        "-m",
        message,
    } });
}

File: apps/cli/git.go
package main

import (
	"errors"
	"os/exec"
)

func getStagedChanges() (string, error) {
	cmd := exec.Command("git", "diff", "--cached")
	stdout, err := cmd.Output()

	if err != nil {
		return "", err
	}

	if len(stdout) == 0 {
		return "", errors.New("no staged changes found")
	}

	return string(stdout), nil
}

func commit(message string) error {
	cmd := exec.Command("git", "commit", "-m", message)
	return cmd.Run()
}
